include/seastar/core/shared_mutex.hh

```diff
@@ -234,6 +237,44 @@ with_lock(shared_mutex& sm, Func&& func) noexcept {
     }
 }
 
+/// \brief Construct a RAII-based shared lock corresponding to a given \ref shared_mutex object.
+///
+/// Since constructors cannot be exectued asynchronously, use this function
+/// to shared lock a \ref shared_mutex and construct an std::shared_lock object
+/// that will unlock the mutex when it's being destroyed.
+///
+/// The caller is responsible for keeping the corresponding \ref shared_mutex object
+/// alive at least until the returned lock is destroyed.
+inline future<std::shared_lock<shared_mutex>> get_shared_lock(shared_mutex& mutex) {
+    co_await mutex.lock_shared();
+
+    try {
+        co_return std::shared_lock<shared_mutex>{mutex, std::adopt_lock_t{}};
```

Comment:
std::shared_lock is copyable, calling the murex.lock_shared, but in seastar case, this returns a future.  We should prevent this.

Reply:
It seems it's actually not. Fortunately the standard specifies that both the copy constructor and the copy assignment operator are both deleted ([link](https://eel.is/c++draft/thread.lock.shared#general))

Reply:
yes, sorry, I confused it with another ctor.